torch/_decomp/decompositions.py

```diff
@@ -2146,22 +2146,29 @@ def upsample_bilinear2d(
     out_w = sym_float(output_size[1])
 
     # Calculate horizontal and vertical scaling factor
-    # TODO: Figure out if scales_h/scales_w matters here
-    if out_h > 1:
-        if align_corners:
-            h_scale_factor = (in_h - 1) / (sym_int(out_h) - 1)
+    def compute_scales_value(
+        scale: Optional[float], input_size: int, output_size: int
+    ) -> float:
+        # FIXME: remove magic > 0 after we ensure no models were serialized with
+        # -1 defaults.
+        if scale is not None and scale > 0:
+            return 1 / scale
         else:
-            h_scale_factor = in_h / out_h
-    else:
-        h_scale_factor = 0.0
+            return input_size / output_size
 
-    if out_w > 1:
+    def area_pixel_compute_scale(
+        input_size: int, output_size: int, align_corners: bool, scale: Optional[float]
+    ) -> float:
         if align_corners:
-            w_scale_factor = (in_w - 1) / (sym_int(out_w) - 1)
+            if output_size > 1:
```

Comment:
what happens if align_corners is false and `output_size` is not given? `scale` should not be 0 in this case.

Reply:
This matches the existing logic on the CPP side. Should we fix it there as well, then? Or is it okay to keep the decomp as is given that it's the same logic?

```cpp
template <typename scalar_t>
static inline scalar_t area_pixel_compute_scale(
    int64_t input_size,
    int64_t output_size,
    bool align_corners,
    const c10::optional<double> scale) {
  // see Note [area_pixel_compute_scale]
  if(align_corners) {
    if(output_size > 1) {
      return static_cast<scalar_t>(input_size - 1) / (output_size - 1);
    } else {
      return static_cast<scalar_t>(0);
    }
  } else {
    return compute_scales_value<scalar_t>(scale, input_size, output_size);
  }
}
```

Reply:
Ok as far as I can see in public APIs (`torch.nn.functional.interpolate` and `torch.nn.functional.upsample_bilinear`) one cannot specify both scales and output_size, what would be the way to make a torch call so that both output_size and scales are specified? This decomp is for `.default` overload which requires output size to be specified (function signature implies this also), this means that scales should be None?

Reply:
well if you want to manually trigger the bad case you could call it via torch.ops.aten.blahblah

Reply:
Yeah but my question is why should we support providing both output size and scale factors in decomposition, if this is a clear error that should never happen (and doesn't happen with public APIs, as far as I could tell)? We could make aten function error out in this case too (it doesn't).

Reply:
yeah let's make it error